web/components/PetPostModal/PetPostModal.tsx

```diff
@@ -0,0 +1,608 @@
+import { ArrowBackIcon, CheckIcon } from "@chakra-ui/icons";
+import "react-responsive-carousel/lib/styles/carousel.min.css";
+import {
+  Button,
+  Center,
+  Flex,
+  Modal,
+  ModalContent,
+  Stack,
+  Text,
+  Wrap,
+  WrapItem,
+} from "@chakra-ui/react";
+import ImageSlider from "./ImageSlider";
+
+const PetPostModal: React.FC<{
+  isOpen: boolean;
+  onOpen: () => void;
+  onClose: () => void;
+}> = ({ isOpen, onOpen, onClose }) => {
+  return (
+    <Modal isOpen={isOpen} size={"full"} onClose={onClose}>
+      <ModalContent>
+        <Stack direction="column" display={["none", "flex"]}>
+          <Button
+            onClick={onClose}
+            bgColor="white"
+            _hover={{ bgColor: "white" }}
+            width="10%"
```

Comment:
Leave width unspecified - typically we want to avoid using percentages for non-layout elements (like anything other than what can be boiled down to a div).